text/0006-BundleContext-Install-API.md

```diff
@@ -0,0 +1,136 @@
+
+```c++
+  /**
+   * Insert bundles from a container into persistent storagedata.
+   *
+   * This is a simpler version of the InstallBundles api that only deals with one bundle at a time.
+   * 
+   * @param resCont The container for the bundle data and resources.
+   * @param topLevelEntries The top level entries in the container to be inserted as bundle archives.
+   * @return A list of BundleArchive instances representing the installed bundles.
+   */
+  using ManifestT = cppmicroservices::AnyMap;
+  virtual std::shared_ptr<BundleArchive> InsertArchive(const  std::shared_ptr<BundleResourceContainer>& resCont
+                                                       , const std::string& topLevelEntry
+                                                       , const ManifestT&) = 0;
+
+```
+
+* Combined BundleStorage::InsertBundleLib and BundleStorage::InsertArchives
+  * InsertBundleLib was a wrapper around InsertArchives
+    * It created a BundleResourceContainer for the location and called InsertArchives
+    * Now the BundleResourceContainer is created if needed in BundleRegistry::GetAlreadyInstalldBundlesAtLocation and then passed in to BundleStorage::InsertArchive.
+
+### BundleArchive
+
+* Refactor and remove Data class simplifying design.
+* Add BundleManifest and constructor argument
+  * Call BundleManifest::Parse from BundleArchive in the event that a manifest is not passed in at construction
+* Add new **GetManifest()** method which does a lazy loadof the manifest from the archive if and only if we don't already have the manifest.
+
+### BundleResourceContainer
+
+* Add **manifest** argument to constructor
+  * If the **manifest** is not empty, use it to initialize the **m_SortedToplevelDirs**, and leave **m_IsContainerOpen** set to false
+    * if **manifest** is empty, no manifest injection is happening, so open up the archive and read the manifest from there, set set **m_IsContainerOpen** to true.
+
+### BundleStorage(Memory)
+
+* Refactor
+  * Remove **InsertArchives()** method and move loop to caller in **BundleRegistry::Install0**
+
+### Other Changes
+
+* Various code cleanups (getting rid of "using std", exraneous cout output, etc.)
+
+## Test Cases
+
+- Install bundle manifest at a location, and then retrieve bundle and check manifest
+- Install bundle manifest for a test bundle that exists. Then get the service, call a method through the service interface and made sure that it was successfully started.
+- Install bundle manifest for multiple bundles that exist on disk and then start each one and verify that they have been properly instantiated.
+- Ensure Install bundle manifest for bundle already installed is ignored
```

Comment:
not sure what is meant by "ignored". A test should guarantee that a bundle installed more than once always returns the originally installed bundle to the caller and does not create multiple entries in the bundle registry for the same bundle.

Reply:
Updated description of this test case to match your wording.